### PR TITLE
Issue 269: Sentence incomplete

### PR DESCRIPTION
**Pull request by JRustyHaner**
#269 

-Reproduced bug by adding text and a second blank to testStim.json and dropping the database in postgres and re-initializing. This concatenated clozeText and text on the first run of a given card. Unsure if related.

Fix: reordering the condition statement on line 1225 in card.js

### CODE REVIEW

**Comment by JRustyHaner** [comment]
@andrewtackett this fixed the issue with two blanks, but not three.

**Comment by JRustyHaner** [comment]
image

- Reproduced exact issue

**Comment by JRustyHaner** [comment]
Behavior different error during practice:
image

Reproduced during test:
image

**Comment by JRustyHaner** [comment]
Text and ClozeText are put into the save display variable, having them both displayed in the html causes this issue. Tested 10/17 1:22PM

**Review by andrewtackett — COMMENTED** [review]
It would seem your solution doesn't do underlining for the second time the cloze answer is presented in the question. Also there's a random import in there at the top of unitEngine for some reason.

**Comment by JRustyHaner** [comment]
@andrewtackett

**Review by andrewtackett — CHANGES_REQUESTED** [review]
This still doesn't address underlining the second instance of the cloze word.  Also please remove the reordering of assigning to currentQuestion, clozeText is preferred first for a reason.

**Comment by JRustyHaner** [comment]
image

**Comment by JRustyHaner** [comment]
@andrewtackett

File: mofacts/client/views/experiment/unitEngine.js [hunk 1]
```diff
@@ -126,10 +126,24 @@ function defaultUnitEngine(curExperimentData) {
 
       // eslint-disable-next-line prefer-const
       let clozeQuestionParts = question.split(/([_]+[ ]?)+/);
-      clozeQuestionParts.splice(1, 1);
-      clozeQuestionParts.splice(1, 0, clozeAnswerNoUnderscores.trim());
-      clozeQuestionParts[2] = clozeAnswerOnlyUnderscores + ' ' + clozeQuestionParts[2];
-
+      // clozeQuestionParts.splice(1, 1);
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/unitEngine.js:129]: Shouldn't have commented out code in final versions

File: mofacts/client/views/experiment/unitEngine.js [hunk 2]
```diff
@@ -126,10 +126,24 @@ function defaultUnitEngine(curExperimentData) {
...
+      // clozeQuestionParts.splice(1, 0, clozeAnswerNoUnderscores.trim());
+      // clozeQuestionParts[2] = clozeAnswerOnlyUnderscores + ' ' + clozeQuestionParts[2];
+      // clozeQuestionRemainingParts = clozeQuestionParts.splice(2).join('');
+      // clozeQuestionParts[2] = clozeAnswerOnlyUnderscores + ' ' + clozeQuestionRemainingParts;
+      
+      //Testing
+      for (var i = 0; i < clozeQuestionParts.length; i++)  {
+        console.log('clozeQuestionParts',i,clozeQuestionParts[i]);
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/unitEngine.js:137]: Also should strip testing print statements

File: mofacts/client/views/experiment/unitEngine.js [hunk 3]
```diff
@@ -126,10 +126,24 @@ function defaultUnitEngine(curExperimentData) {
...
+        let clozeAnswerAddedSpaces = "";
+        if(clozeQuestionParts[i].charAt(0) == "_"){
+          clozeQuestionParts[i] = '<u>' + clozeAnswerNoUnderscores + '&nbsp;&nbsp;&nbsp;&nbsp;' + '</u>';
```
Comment (andrewtackett) [inline mofacts/client/views/experiment/unitEngine.js:140]: Why 4 spaces here?

**Review by andrewtackett — CHANGES_REQUESTED** [review]
Looks like injecting the <u></u> worked then?  Just strip out your testing stuff and I think this should be good.

**Comment by JRustyHaner** [comment]
@andrewtackett  removed console log and deprecated code

**Pull request merged**